### PR TITLE
NC-2632: Document execution mode and appearance

### PR DESCRIPTION
**Pull request by worenga**

### CODE REVIEW

**Review by patriciasantaana — APPROVED** [review]

File: content/turnstile/changelog.md [hunk 1]
```diff
@@ -9,6 +9,10 @@ rss: file
 
 ## 2023-02-15
 
+- Added [`[execution]`](/turnstile/get-started/client-side-rendering/#explicitly-render-the-turnstile-widget) and [`[appearance-mode]`](/turnstile/get-started/client-side-rendering/#explicitly-render-the-turnstile-widget).
```
Comment (Le0Developer) [inline content/turnstile/changelog.md]:
```suggestion
- Added [`[execution]`](/turnstile/get-started/client-side-rendering/#explicitly-render-the-turnstile-widget) and [`[appearance]`](/turnstile/get-started/client-side-rendering/#explicitly-render-the-turnstile-widget).
```

File: content/turnstile/get-started/client-side-rendering.md [hunk 2]
```diff
@@ -138,18 +138,20 @@ Once a widget is no longer needed, it can be removed from the page using `turnst
...
 | `cData` | `data-cdata` | A customer payload that can be used to attach customer data to the challenge throughout its issuance and which is returned upon validation. This can only contain up to 255 alphanumeric characters including `_` and `-`.  |
 | `callback` | `data-callback` | A JavaScript callback that is invoked upon success of the challenge. The callback is passed a token that can be validated. |
-| `expired-callback` | `data-expired-callback` | A JavaScript callback that is invoked when the token expires and does not reset the widget. |
-| `timeout-callback` | `data-timeout-callback` | A JavaScript callback that is invoked when the challenge expires and resets the widget. |
 | `error-callback` | `data-error-callback` | A JavaScript callback that is invoked when there is a network error. |
+| `execution-mode` | `data-execution-mode` | Controls when to obtain the token of the widget can be on `render` (default) or on `execute`), see also Execution Modes.
```
Comment (Le0Developer) [inline content/turnstile/get-started/client-side-rendering.md]:
```suggestion
| `execution` | `data-execution` | Controls when to obtain the token of the widget can be on `render` (default) or on `execute`), see also Execution Modes.
```

File: content/turnstile/get-started/client-side-rendering.md [hunk 3]
```diff
@@ -159,3 +161,23 @@ The Turnstile widget can have two different sizes when using the Managed or Non-
...
+By default the parameter is set to `auto` which will automatically make Turnstile obtain a fresh token. Upon regenerating the token automatically the `callback` - if specified - is invoked.
+The visitor can also be instructed to manually obtain a new token by setting the `refresh-expired` parameter to `manual`.
+Additionally, specifying `never` will not result in a regeneration of a token, and the application using Turnstile will be responsible for obtaining a novel Turnstile token.
+
+
+## Execution Modes
+By default Turnstille tokens are obtained for a visitor upon rendering of a widget (even in invisible mode).
+However, in some secenarios an application may want to embed Turnstile, but hold of obtaining the token until a certain point in time.
+This is where execution mode can be used to control when a token is being generated. There are two options: The token is automatically optained upon call of the `render()` function. 
+Alternatively, the token can be obtained after the `render()` function has been called, by invoking the `turnstile.execute(container: string | HTMLElement, jsParams?: RenderParameters)` function separately.
```
Comment (Le0Developer) [inline content/turnstile/get-started/client-side-rendering.md:178]:
```suggestion
Alternatively, the token can be obtained after the `render()` function has been called, by invoking the `turnstile.execute(container: string | HTMLElement | undefined, jsParams?: RenderParameters)` function separately.
```
container is optional

File: content/turnstile/get-started/client-side-rendering.md [hunk 4]
```diff
@@ -159,3 +161,23 @@ The Turnstile widget can have two different sizes when using the Managed or Non-
...
+This detaches the appareance and rendering of a widget from its execution.
+
+## Appearance Modes
+If a widget is visible (i.e. its type is not invisible) the appearance of the widget can be controled via the `appearance-mode` parameter. By default, for visible widget types, the `appearance-mode` is set to `always` and makes widget is always visible. However, if `appearance-mode` is set to execute, the widget will only become visible once it is in the process of obtaining a token (and afterwards). This is helpful in situations where `execute()` is called after `render()`. 
+If `appearance-mode` is set to `interaction-only` the widget will become only visible in cases where an interaction (e.g. clicking a checkbox et al.) is required.
```
Comment (Le0Developer) [inline content/turnstile/get-started/client-side-rendering.md]:
```suggestion
If a widget is visible (i.e. its type is not invisible) the appearance of the widget can be controled via the `appearance` parameter. By default, for visible widget types, `appearance` is set to `always` and makes widget is always visible. However, if `appearance` is set to execute, the widget will only become visible once it is in the process of obtaining a token (and afterwards). This is helpful in situations where `execute()` is called after `render()`. 
If `appearance` is set to `interaction-only` the widget will become only visible in cases where an interaction (e.g. clicking a checkbox et al.) is required.
```

**Review by Le0Developer — COMMENTED** [review]
The parameter is just `appearance` and `execution` without the `-mode` suffix.

Additionally, the changelog date has been the date of documentation so far, not the date when the feature was added.
(also pretty please prefix the PR title with [`[turnstile]`](https://github.com/cloudflare/cloudflare-docs/blob/production/CONTRIBUTING.md#pull-request-guidelines) so it's easier to find)

**Pull request merged**